spidermon/contrib/validation/cerberus/validator.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+from collections.abc import Mapping
+
+from cerberus.validator import (
+    Validator as cerberus_validator,
+    DocumentError,
+    SchemaError,
+)
+from spidermon.contrib.validation.validator import Validator
+from .translator import CerberusMessageTranslator
+
+
+class CerberusValidator(Validator):
+    default_translator = CerberusMessageTranslator()
+    name = "Cerberus"
+
+    def __init__(self, schema, translator=None, use_default_translator=True):
+        super(CerberusValidator, self).__init__(
+            translator=translator, use_default_translator=use_default_translator
+        )
+        if schema is not None:
+            self._schema = schema
+        elif self._schema is None:
+            if isinstance(self._schema, Mapping):
+                self._schema = {}
+            else:
+                raise SchemaError("Validation schema missing")
+
+    def _validate(self, data, strict=False):
+        if data is None:
+            raise DocumentError("Data is missing")
+        try:
+            validator = cerberus_validator(self._schema)
+            validator.validate(data)
+            for field_name, error in validator.errors.items():
+                self._add_errors({field_name: error})
+        except SchemaError:
```

Comment:
You are raising `SchemaError` in `__init__`, so the validator will not be instantiated if the schema is not valid, so I believe it is unnecessary to catch the exception here as well (only if it is possible to get a SchemaError for a reason different for the one you check in __init__.

Reply:
Yes, Renne both errors are from the same error class apparently. That's why 2 catch, raise.
`>>> cerberus.schema.SchemaError: validation schema missing` and 
`>>> cerberus.schema.SchemaError: {'a': ['must be of dict type']}`

**Reasons**
- We am raising `SchemaError` from `__init__` because we are importing SchemaError from Cerberus , hence it only make to raise it when no schema is given. 

- On the other hand, we are catching the `SchemaError`, to raise ValueError instead for uniformity.